table/block.cc

```diff
@@ -240,8 +240,8 @@ class Block::Iter : public Iterator {
       CorruptionError();
       return false;
     } else {
-      key_.resize(shared);
-      key_.append(p, non_shared);
+      key_.resize(shared + non_shared);
```

Comment:
The new way can end up writing twice: once the resize() will fill any added bytes with zeroes, and then the memcpy below will overwrite.

Let's just stick with the current behavior.  Yes, it pays a few cycles more in append(), but that cost is not linear in number of added bytes.